### DISCUSSION TITLE
Connect To Smart Contract using Expo

### DISCUSSION THREAD

**Discussion by Ronexlemon** [WalletConnect/walletconnect-monorepo#2921] [Developer Support]
this is the error am getting "err Object {
  "code": -32000,
  "message": "JSON Parse error: Unexpected end of input",
}
" when i try to connect to a smart contract" AND this the code to connect " if (web3Provider) {
        
      const {chainId} = await web3Provider.getNetwork();
      const sign2 =  web3Provider.getSigner();
      const contract = new Contract(
        ChainLinkContract,
        ierc20,
        sign2
      );
    
       await contract.approve(FantomContractAddress,amount);"

**Comment by crypblizz8** [comment]
Hey there you mentioned Expo which I assume is for React Native,

Can you provide more context + information on your setup. Thanks

**Comment by Ronexlemon** [comment]
I have already  connected my react native app using wallet connect  but how
will i connect smartcontract and call one of its function

**Comment by Ronexlemon** [comment]
I just want a documentation of how i will use the ethereum provider then
call a smartcontract
